PORTING.md

```diff
@@ -4,6 +4,38 @@ Porting applications to use zlib-ng
 Zlib-ng can be used/compiled in two different modes, that require some
 consideration by the application developer.
 
+Changes from zlib affecting native and compat modes
+---------------------------------------------------
+Zlib-ng is not as conservative with memory allocation as Zlib is.
+
+Where Zlib's inflate will allocate a lower amount of memory depending on
+compression level and window size, zlib-ng will always allocate the maximum
+amount of memory and possibly leave parts of it unused.
+Zlib-ng's deflate will however allocate a lower amount of memory depending
+on compression level and window size.
+
+Zlib-ng also allocates one "big" buffer instead of doing multiple smaller
+allocations. This is faster, can lead to better cache locality and reduces
+space lost to alignment padding.
+
+At the time of writing, by default zlib-ng allocates the following amounts
+of memory on a 64-bit system (except on S390x that requires ~4KiB more):
+-  Deflate: 350.272 Bytes
```

Comment:
This is for the struct only or for the window and pending buffers?

Reply:
This is all we allocate, everything.
```
window is 65536 bytes, offset 0, padded 0
prev is 65536 bytes, offset 65536, padded 0
head is 131072 bytes, offset 131072, padded 0
pending is 81920 bytes, offset 262144, padded 0
state is 6080 bytes, offset 344064, padded 0
alloc is 56 bytes, offset 350144, padded 0
Buffer alloc is 350272 bytes, offset 0, padded 8
```

Reply:
I see, I misread the period since I am so used to seeing a comma..